src/app/config/projectsJsonData.ts

```diff
@@ -1,128 +1,116 @@
-    }
+  constructor(templatePath: string) {
+    let jsonData = fs.readFileSync(templatePath + this.m_projectJsonDataFile);
+    this.m_projectJsonData = JSON.parse(jsonData.toString());
+  }
 
-    isValidInput(input: string, isHostParam: boolean)
-    {
-      if (isHostParam)
-      {
-        for (let key in this.m_projectJsonData.hostTypes)
-        {
-           if (_.toLower(input) == key){
-             return true;
-           }
-          }
-          return false;
+  isValidInput(input: string, isHostParam: boolean) {
+    if (isHostParam) {
+      for (let key in this.m_projectJsonData.hostTypes) {
+        if (_.toLower(input) == key) {
+          return true;
         }
-        else{
-          for (let key in this.m_projectJsonData.projectTypes)
-          {
-            if (_.toLower(input) == key){
-              return true;
-            }
-          }
-          return false;
-        }
-    }
-
-    getProjectDisplayName(projectType: string){
-      return this.m_projectJsonData.projectTypes[_.toLower(projectType)].displayname;
-    }
-    
-    getParsedProjectJsonData()
-    {
-      return this.m_projectJsonData;
+      }
+      return false;
     }
-    
-    getProjectTemplateNames()
-    {
-      let projectTemplates : string[] = [];
-      for (let key in this.m_projectJsonData.projectTypes)
-      {
-        projectTemplates.push(key);
+    else {
+      for (let key in this.m_projectJsonData.projectTypes) {
+        if (_.toLower(input) == key) {
+          return true;
+        }
       }
-      return projectTemplates;
+      return false;
     }
-    
-    projectBothScriptTypes (projectType: string)
-    {
-      return this.m_projectJsonData.projectTypes[_.toLower(projectType)].templates.javascript != undefined && this.m_projectJsonData.projectTypes[_.toLower(projectType)].templates.typescript != undefined;
+  }
+
+  getProjectDisplayName(projectType: string) {
+    return this.m_projectJsonData.projectTypes[_.toLower(projectType)].displayname;
+  }
+
+  getParsedProjectJsonData() {
+    return this.m_projectJsonData;
+  }
+
+  getProjectTemplateNames() {
+    let projectTemplates: string[] = [];
+    for (let key in this.m_projectJsonData.projectTypes) {
+      projectTemplates.push(key);
     }
+    return projectTemplates;
+  }
+
+  projectBothScriptTypes(projectType: string) {
+    return this.m_projectJsonData.projectTypes[_.toLower(projectType)].templates.javascript != undefined && this.m_projectJsonData.projectTypes[_.toLower(projectType)].templates.typescript != undefined;
+  }
 
-    getHostTemplateNames()
-    {
-      let hosts : string[] = [];
-      for (let key in this.m_projectJsonData.hostTypes)
-      {
+  getHostTemplateNames(isSsoProject: boolean) {
+    let hosts: string[] = [];
+    for (let key in this.m_projectJsonData.hostTypes) {
+      if (isSsoProject) {
+        if (key === 'excel' || key === 'word' || key === 'powerpoint') {
```

Comment:
SSO project excludes Outlook, right?

Reply:
Might be better to say:

```
if (key == 'outlook') {
     // exclude Outlook for SSO project, so don't add host 
} else {
   host.push...
}

Reply:
yes that's correct for now. We tried to add it really late but were hitting some issues that reduced our confidence that we are ready to support it

Reply:
You could combine two host.push lines so there is only one where the if is `if (isSsoProject && (key === 'outlook')) {`

Reply:
Just to be clear we don't support OneNote or Project either

Reply:
It's more than just Outlook, also OneNote and Project. I think ideally the data should drive which templates are supported for which hosts. Let's put that down for a future improvement and you can leave the code for now.